cmake/modules/SwiftComponents.cmake

```diff
@@ -127,42 +127,30 @@ function(swift_is_installing_component component result_var_name)
   endif()
 endfunction()
 
-# swift_install_in_component(<COMPONENT NAME>
-#   <same parameters as install()>)
+# swift_install_in_component(<same parameters as install()>)
 #
 # Executes the specified installation actions if the named component is
 # requested to be installed.
 #
 # This function accepts the same parameters as install().
-function(swift_install_in_component component)
-  precondition(component MESSAGE "Component name is required")
+function(swift_install_in_component)
+  cmake_parse_arguments(
+      ARG # prefix
+      "" # options
+      "COMPONENT" # single-value args
+      "" # multi-value args
+      ${ARGN})
```

Comment:
Usually this is nicer with variables for the individual options rather than comments, but, this should be short lived, so meh.